hivemind/utils/grpc.py

```diff
@@ -207,6 +248,18 @@ def serialize_torch_tensor(tensor: torch.Tensor, compression_type=CompressionTyp
             size=array.shape,
             dtype=array.dtype.name,
             requires_grad=tensor.requires_grad)
+    elif compression_type == CompressionType.QUANTILE_8BIT:
+        assert tensor.dtype == torch.float32
+
+        quantized, lookup = quantile_encode_torch_approx(tensor.detach(), 8)
+        data = b''.join((lookup.numpy().tobytes(), quantized.numpy().astype(np.uint8).tobytes()))
```

Comment:
Casting to ByteTensor before converting to np.array is probably more efficient

Reply:
it is not, .numpy() is a no-op for cpu tensors